app/models/solidus_stripe/setup_intent.rb

```diff
@@ -3,11 +3,26 @@
 module SolidusStripe
   class SetupIntent < Spree::Base
     belongs_to :order, class_name: 'Spree::Order'
+    belongs_to :payment_method, class_name: 'SolidusStripe::PaymentMethod'
 
-    def stripe_setup_intent(payment_method)
+    before_create :create_stripe_intent, unless: :stripe_setup_intent_id?
```

Comment:
Now that we removed the call to the custom `find_or_create_setup_intent_for_order` from the payment method, we are not creating this explicitly. Instead, we are relying on the AR interface in order to call `find_or_create_by`. They do the same thing, it's just more Rails way, and moved to the right place. 


Also, to be clear, we are creating the stripe intent in Stripe, and just saving its id on the internal intent. Do you think it's a matter of finding a better name?

Reply:
The issue is that, in my view, mixing persistence with network calls is not a good idea. I'd prefer a clear and explicit interface. Otherwise, we risk not handling network errors and leaving the db on an inconsistent state, besides probably slowing down some unit tests in the future.

Reply:
That's a very valid point. About slowing down testing, I think it won't happen if we have the factory setting the `intent_id` but it could happen anyway in some cases, where we are not explicitly using factories when creating intents.

In practice, do you suggest to explicitly doing both things from the caller method, defining a class method on the Setup/PaymentIntent that takes care of both things, or creating a service object? I'm fine with all of them, and also with a temporary solution, because maybe we will have the perfect place for this code after the work we are doing these days.

Reply:
We (me and @waiting-for-dev) rearranged the code to avoid the AR callback and also avoid AR operations in the view 🙌